sunpy/net/hek/tests/test_utils.py

```diff
@@ -0,0 +1,175 @@
+import copy
+
+import numpy as np
+import pytest
+
+import astropy.table
+import astropy.time
+import astropy.units as u
+import astropy.utils
+import astropy.utils.masked
+from astropy.coordinates import SkyCoord
+from astropy.tests.helper import assert_quantity_allclose
+
+from sunpy.coordinates import (
+    Heliocentric,
+    HeliographicCarrington,
+    HeliographicStonyhurst,
+    Helioprojective,
+    get_earth,
+)
+from sunpy.net.hek.utils import (
+    _map_chain_code_columns_to_coordinates,
+    _map_columns_to_quantities,
+    _map_event_coord_columns_to_coordinates,
+    _parse_unit,
+)
+
+
+@pytest.mark.parametrize(('input_unit', 'expected_unit'), [
+    ('DN/sec/pixel', u.DN/(u.pix*u.s)),
+    ('ergs per cubic centimeter', u.erg/u.cm**3),
+    ('HMI pixel', 0.5*u.arcsec/u.pixel),
+    ('HMI pixels', 0.5*u.arcsec/u.pixel),
+    ('ma/m2', u.milliampere/u.m**2),
+    ('deg deg', u.deg),
+    ('arcsec, arcsec', u.arcsec),
+    ('arcsec,arcsec', u.arcsec),
+])
+def test_parse_unit(input_unit, expected_unit):
```

Comment:
I feel like this function could use some more testing. There looks to be some branches which are maybe not getting hit. Also what happens if you throw some unexpected input at this thing?

Reply:
Define unexpected input. Do you mean something that will not parse as an astropy unit that is not covered by the aliases here? If so, then the answer is "the same thing that happens when you stick in something unexpected to u.Unit".

Reply:
I added a test that makes sure that is the case.